File: NirvanaSMS.Api.CSharp/Core/Base/BlackListBase.cs
using NirvanaSMS.Api.Business.Interface;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;
using NirvanaSMS.Api.Core.Interface;
using NirvanaSMS.Api.Core.Exceptions;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class BlackListBase : IBlackList
    {
        public virtual ProcessResult Add(ApiUser apiUser, PhoneNumber phoneNumber)
        {
            Dictionary<string, string> getItems = new Dictionary<string, string>();
            if (phoneNumber.IsValid)
            {
                getItems.Add("username", apiUser.UserName);
                getItems.Add("password", apiUser.ApiPassword);
                getItems.Add("gsm", phoneNumber.Number);
                HttpRequest request = new HttpRequest(Constants.BlackListAddUrl,getItems,Enums.RequestMethod.Get);
                string requestResult = request.GetResponse();
                return new ProcessResult(requestResult);
            }
            throw new NonValidPhoneNumberException("Numara geçerli bir formata sahip değil.");
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/BlackListBase.java
package org.nirvanasms.api.core.base;

import java.util.HashMap;
import java.util.Map;

import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.entity.PhoneNumber;
import org.nirvanasms.api.core.exceptions.NonValidPhoneNumberException;
import org.nirvanasms.api.core.interfaces.IBlackList;
import org.nirvanasms.api.core.utility.Constants;
import org.nirvanasms.api.core.utility.Enums;
import org.nirvanasms.api.core.utility.Enums.RequestMethod;
import org.nirvanasms.api.core.utility.HttpRequest;
import org.nirvanasms.api.core.utility.ProcessResult;

public class BlackListBase implements IBlackList {

	@SuppressWarnings("rawtypes")
	@Override
	public ProcessResult<?> add(ApiUser apiUser, PhoneNumber phoneNumber) throws NonValidPhoneNumberException {

		Map<String, String> getItems = new HashMap<String, String>();
		if (phoneNumber.isValid()) {
			getItems.put("username", apiUser.getUserName());
			getItems.put("password", apiUser.getApiPassword());
			getItems.put("gsm", phoneNumber.getNumber());
			HttpRequest request = new HttpRequest(Constants.BlackListAddUrl,getItems,RequestMethod.Get);
			String requestResult = request.getResponse();
			return new ProcessResult(requestResult);
		}
		throw new NonValidPhoneNumberException("Numara geçerli bir formata sahip değil.");
	}

}
